util/design/lib/OtpMemMap.py

```diff
@@ -201,29 +216,39 @@ def _validate_mmap(config):
         if offset > check_int(part["offset"]) + check_int(part["size"]):
             log.error("Not enough space in partitition "
                       "{} to accommodate all items. Bytes available "
-                      "= {}, bytes requested = {}".format(
-                          part["name"], part["size"],
-                          offset - part["offset"]))
+                      "= {}, bytes allocated to items = {}".format(
+                          part["name"], part["size"], offset - part["offset"]))
             exit(1)
 
         offset = check_int(part["offset"]) + check_int(part["size"])
 
+        part_index.setdefault(part['name'], {
+            'index': num_part,
+            'items': item_index
+        })
+        num_part += 1
```

Comment:
As with the num_items loop, this might be easier if you looped over `enumerate(config['partitions'])`. Apart from anything else, it lets you keep the update of num_part next to the loop itself, which makes things a bit easier to follow.

Reply:
again, same story as above: the variable is used outside of the loop context later on

Reply:
Hmm. I think I'd still go for the enumerate (in both cases) and then replace e.g. the later `num_part` with `len(config['partitions'])`. But I don't feel strongly.

Reply:
ok I've switched over to your suggestion. it does improve readability a bit and makes the code a tick shorter.